File: Sample/app/src/main/java/net/nend/sample/NativePagerFragment.java
package net.nend.sample;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

public class NativePagerFragment extends Fragment {

    OnAdListener mCallback;

    public interface OnAdListener {
        void onAdRequest(View view, int position);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        Activity activity = getActivity();
        try {
            mCallback = (OnAdListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnHeadlineSelectedListener");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.native_fragment, container, false);
        RelativeLayout adLayout = (RelativeLayout) view.findViewById(R.id.ad);

        int position = getArguments().getInt("position");
        mCallback.onAdRequest(adLayout, position);

        return view;
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/nativead/NativePagerFragment.kt
package net.nend.sample.kotlin.nativead

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import androidx.fragment.app.Fragment
import net.nend.sample.kotlin.R

class NativePagerFragment : Fragment() {

    internal lateinit var callback: OnAdListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity = activity
        try {
            callback = activity as OnAdListener
        } catch (e: ClassCastException) {
            throw ClassCastException(activity.toString() + " must implement OnHeadlineSelectedListener")
        }
    }

    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return View.inflate(context, R.layout.native_fragment, null).apply {
            val adLayout = findViewById<RelativeLayout>(R.id.ad)
            callback.onAdRequest(adLayout, requireArguments().getInt("position"))
        }
    }

    interface OnAdListener {
        fun onAdRequest(view: View, position: Int)
    }
}